### PR TITLE
drop flux-jobspec command

### PR DESCRIPTION
**Pull request by garlick**
`flux jobspec srun` really isn't needed anymore now that `flux mini` has the `--dry-run` option for capturing generated jobspec.  This converts tests that still use `flux jobspec` over to `flux mini` and then removes the command and its test.

flux-framework/flux-sched#884 would need to go in before this can be merged.

### CODE REVIEW

**Review by chu11 — APPROVED** [review]
LGTM, always good to get rid of unused code!

**Comment by garlick** [comment]
@mergify-io rebase

**Comment by garlick** [comment]
@mergifyio rebase

**Comment by garlick** [comment]
One builder hit #3546, so restarting CI.

File: t/t3100-flux-in-flux.t [hunk 1]
```diff
@@ -64,15 +64,15 @@ test_expect_success "instance-level attribute = 0 in test instance" '
...
+		flux getattr instance-level >level1.out &&
 	echo 1 >level1.exp &&
 	test_cmp level1.exp level1.out
 '
 
 test_expect_success "instance-level attribute = 2 in second subinstance" '
-        flux mini run flux start \
-	    flux mini run flux start ${ARGS} \
+	flux mini run flux start \
+		flux mini run flux start ${ARGS} \
```
Comment (grondo) [inline t/t3100-flux-in-flux.t:72]: I'm fine with these changes, but in general I think indenting a couple spaces in a line continuation (after the initial tab) should be fine in sharness tests and not considered inconsistent use of tabs/spaces going forward. Using a smaller indent at a line continuation can help avoid lines getting too long in many situations. (I do agree, however, that the initial indent should always be a tab in sharness tests.)

File: t/t0022-jj-reader.t [hunk 2]
```diff
@@ -8,23 +8,25 @@ jj=${FLUX_BUILD_DIR}/t/sched-simple/jj-reader
 y2j="flux python ${SHARNESS_TEST_SRCDIR}/jobspec/y2j.py"
 
 test_expect_success HAVE_JQ 'jj-reader: unexpected version throws error' '
-	flux jobspec srun hostname | jq ".version = 2" >input.$test_count &&
```
Comment (grondo) [inline t/t0022-jj-reader.t:11]: In commit message: missing word(?) "the flux jobspec _command_ is being removed"?

**Review by grondo — APPROVED** [review]
LGTM! 

Just one possible commit message typo inline, also there is one more reference to `flux jobspec` in the code [here](https://github.com/grondo/flux-core/blob/a95d42e3e83ee01026763793d0fb29143db3913f/src/shell/test/jobspec.c#L34), but it is in a comment so not a big deal.

File: t/t3100-flux-in-flux.t (hunk 1 above)
Comment (garlick) [reply t/t3100-flux-in-flux.t:72]: OK that is reasonable.

**Comment by garlick** [comment]
Hmm 32-bit builder failed valgrind test due to the instance not exiting with success:
```
2021-11-11T19:24:53.4493011Z ==663257== For counts of detected and suppressed errors, rerun with: -v
2021-11-11T19:24:53.4493898Z ==663257== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 33033 from 17)
2021-11-11T19:24:53.4494820Z flux-start: 0 (pid 663257) exited with rc=1
2021-11-11T19:24:53.4495698Z not ok 1 - valgrind reports no new errors on 2 broker run
```
Restarting to see if that was a fluke.

**Comment by garlick** [comment]
I'm going to go ahead and set MWP, thanks!

**Pull request merged**